src/olympia/signing/views.py

```diff
@@ -113,6 +114,39 @@ def check_throttles(self, request):
             return
         super().check_throttles(request)
 
+    # When DRF 3.12 is released, we can remove the custom check_permissions()
+    # and permission_denied() as it will contain the fix for
+    # https://github.com/encode/django-rest-framework/issues/7038
+    def check_permissions(self, request):
+        """
+        Check if the request should be permitted.
+        Raises an appropriate exception if the request is not permitted.
+
+        (Lifted from DRF, but also passing the code argument down to the
+        permission_denied() call if that property existed on the failed
+        permission class)
+        """
+        for permission in self.get_permissions():
+            if not permission.has_permission(request, self):
+                self.permission_denied(
+                    request, message=getattr(permission, 'message', None),
+                    code=getattr(permission, 'code', None),
+                )
+
+    def permission_denied(self, request, message=None, code=None):
+        """
+        If request is not permitted, determine what kind of exception to raise.
+
+         (Lifted from DRF, but also passing the optional code argument to
+          the PermissionDenied exception)
+        """
+        if request.authenticators and not request.successful_authenticator:
+            raise exceptions.NotAuthenticated()
+        # If I can't figure out how to properly get the code without breaking
+        # stuff, maybe I can cheat and return an entirely different class here
```

Comment:
s/I/we

Reply:
Ooops that comment shouldn't be there anymore since I did figure out a way to make the exception handler work. I'll remove it.